addons/binding/org.openhab.binding.sinope/src/main/java/org/openhab/binding/sinope/handler/SinopeThermostatHandler.java

```diff
@@ -0,0 +1,261 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.sinope.handler;
+
+import java.io.IOException;
+import java.net.UnknownHostException;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.sinope.SinopeBindingConstants;
+import org.openhab.binding.sinope.internal.config.SinopeConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ca.tulip.sinope.core.SinopeDataReadRequest;
+import ca.tulip.sinope.core.SinopeDataWriteRequest;
+import ca.tulip.sinope.core.appdata.SinopeHeatLevelData;
+import ca.tulip.sinope.core.appdata.SinopeOutTempData;
+import ca.tulip.sinope.core.appdata.SinopeRoomTempData;
+import ca.tulip.sinope.core.appdata.SinopeSetPointModeData;
+import ca.tulip.sinope.core.appdata.SinopeSetPointTempData;
+import ca.tulip.sinope.core.internal.SinopeDataAnswer;
+import ca.tulip.sinope.util.ByteUtil;
+
+/**
+ * The {@link SinopeThermostatHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Pascal Larin - Initial contribution
+ */
+public class SinopeThermostatHandler extends BaseThingHandler {
+
+    private static final int DATA_ANSWER = 0x0A;
+
+    private Logger logger = LoggerFactory.getLogger(SinopeThermostatHandler.class);
+
+    private SinopeGatewayHandler gatewayHandler;
+
+    private byte[] deviceId;
+
+    public SinopeThermostatHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        SinopeGatewayHandler gateway = this.gatewayHandler;
+        gateway.stopPoll();
```

Comment:
Do you really have to cancel the polling? Note that this method is called with REFRESH commands as well (btw, you do not seem to handle them?), which can mean that you have multiple calls within a second - which could case issues/delays when cancelling and recreating the polling job all the time.